pkg/validations/cluster.go

```diff
@@ -114,3 +118,132 @@ func ValidateManagementClusterBundlesVersion(ctx context.Context, k KubectlClien
 
 	return nil
 }
+
+// ValidateEksaVersion ensures that the version matches EKS-A CLI.
+func ValidateEksaVersion(ctx context.Context, k KubectlClient, cliVersion string, workload *cluster.Spec) error {
+	v := workload.Cluster.Spec.EksaVersion
+
+	if cliVersion == "" {
+		cliVersion = version.Get().GitVersion
+	}
+
+	if v != nil {
+		parsedVersion, err := semver.New(string(*v))
+		if err != nil {
+			return fmt.Errorf("parsing cluster eksa version: %v", err)
+		}
+
+		parsedCLIVersion, err := semver.New(cliVersion)
+		if err != nil {
+			return fmt.Errorf("parsing eksa cli version: %v", err)
+		}
+
+		if !parsedVersion.SamePatch(parsedCLIVersion) {
+			return fmt.Errorf("cluster's eksaVersion does not match EKS-A")
+		}
+	}
+	return nil
+}
+
+// ValidateEksaVersionSkew ensures that upgrades are sequential by CLI minor versions.
+func ValidateEksaVersionSkew(ctx context.Context, k KubectlClient, cluster *types.Cluster, workload *cluster.Spec) error {
+	c, err := k.GetEksaCluster(ctx, cluster, cluster.Name)
+	if err != nil {
+		return err
+	}
+
+	v := c.Spec.EksaVersion
+
+	// try getting version from bundle's eksa component's cluster controller image
+	if v == nil {
+		version, err := getVersionFromBundle(ctx, k, c.Spec.BundlesRef, cluster)
```

Comment:
i would vote to not do this
if the management cluster doesn't yet have a eksaVersion, then we don't validate this